parsec/core/fs/workspacefs/workspacefs.py

```diff
@@ -186,12 +186,55 @@ def workspace_name(self) -> str:
     async def move(self, source: AnyPath, destination: AnyPath):
         source = FsPath(source)
         destination = FsPath(destination)
+        real_destination = destination
         if source.parent == destination.parent:
             return await self.rename(source, destination)
-        # TODO - reference implementation:
-        # https://github.com/python/cpython/blob/3.7/Lib/shutil.py#L525
+        try:
+            if await self.is_dir(destination):
+                real_destination = destination.joinpath(source.name)
+                if await self.path_info(real_destination):
+                    raise FileExistsError
+        # If real_destination is not found, we can continue
+        except FileNotFoundError as e:
```

Comment:
This `try/except` is a bit wide, could we shrink it?

Reply:
It responds to the need of having real_destination referencing either the destination path if it didn't already exist, or a new entry with the same name as source, but inside the destination directory. I updated the comment, is it clear enough or should I try to implement it another way?